src/weapons.cpp

```diff
@@ -836,13 +846,13 @@ int32_t WeaponDistance::getElementDamage(const Player* player, const Creature* t
 	return -normal_random(minValue, static_cast<int32_t>(maxValue * player->getVocation()->distDamageMultiplier));
 }
 
-int32_t WeaponDistance::getWeaponDamage(const Player* player, const Creature* target, const Item* item,
-                                        bool maxDamage /*= false*/) const
+int32_t WeaponDistance::getWeaponDamage(std::shared_ptr<const Player> player, std::shared_ptr<const Creature> target,
+                                        std::shared_ptr<const Item> item, bool maxDamage /*= false*/) const
 {
 	int32_t attackValue = item->getAttack();
 
 	if (item->getWeaponType() == WEAPON_AMMO) {
-		Item* weapon = player->getWeapon(true);
+		auto weapon = player->getWeapon(true);
```

Comment:
Could this be `const auto`
```suggestion
		cont auto weapon = player->getWeapon(true);
```